### DISCUSSION TITLE
SPI's port forwarding and scan prevention.

### DISCUSSION THREAD

**Discussion by killgapwalker** [strongswan/strongswan#426] [General]
Somebody know how to change spi's port's and make after scan prevention. ?

**Comment by Thermi** [comment]
There's no such thing as an SPI port and standard compliant IPsec implementations drop IPsec packets with unknown SPIs and also drop invalid, unauthenticated, incorrectly authenticated, and packets that fail to decapsulate generally.

**Reply by killgapwalker** [reply]
SPI's is a switch without ports ?

**Reply by Thermi** [reply]
SPI is security parameter index. https://en.wikipedia.org/wiki/Security_Parameter_Index

**Reply by killgapwalker** [reply]
Okay ,then i need to change ports on strongswan client to preroute behind nat .

**Reply by Thermi** [reply]
What do you hope to achieve?

**Reply by killgapwalker** [reply]
secured ipsec with insolente.

**Reply by Thermi** [reply]
IPsec itself is already a security solution. What do you mean with "insolente"?

**Comment by killgapwalker** [comment]
Okay , then i need double solution which will hide my 500,4500 port and his destination in other port. Bcs i'm watching how to move 443,80,53 ... This solution will help me obfuscate my internet connection with dedicated server.

**Reply by Thermi** [reply]
There is no security benefit to that. It only adds obscurity and at most cleans up your logs because simple automated scanners won't try other ports other than the IANA assigned default ones.

**Reply by Thermi** [reply]
At worst, the operational expense of setting up that obscurity and maintaining it throughout the life of the whole setup is time and money not spent on improving the security of the setup and/or making money. Not a good trade.
